c11threads.h

```diff
@@ -37,11 +77,34 @@ Main project site: https://github.com/jtsiomb/c11threads
 #endif
 
 /* types */
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+typedef unsigned long thrd_t;
+#pragma pack(push, 8)
+typedef struct {
+	void *debug_info;
+	long lock_count;
+	long recursion_count;
+	void *owning_thread;
+	void *lock_semaphore;
+	void *spin_count;
+} mtx_t;
+#pragma pack(pop)
+#ifndef C11THREADS_SUPPORT_WINNT_OLDER_THAN_VISTA
```

Comment:
I don't think there's much point in adding that ifdef. What I meant by supporting older versions was that at some point in the future, it would be nice to either change the code to not use any vista APIs at all, or add detection and fallbacks for older win32. Until then it suffices to add a note to the README that the current win32 implementation requires vista or later.

Reply:
I see, I misunderstood. I still think it would be nice to provide the library without `cnd_...()` and `call_once()` for those on older Windows versions. From what I've gathered, `call_once()` can be replaced with some atomic functions and polling and I could easily implement that. But the condition variable functions are problematic. We simply don't have atomic wait-and-release operations prior to Vista.

Reply:
You're not wrong. I suggest picking a less verbose name for it then like C11THREADS_OLD_WINAPI, or maybe something more directly related to what you're disabling, like C11THREADS_NO_COND / NO_ONCE + documentation about when to define them?

While there are no  condvar-equivalent before vista, I have implemented condvars on windows once upon a time, with a combination of a critical section, a semaphore, an event and SignalObjectAndWait... I could bundle up the code and attach it here if you're interested, but the comment at the top says: 

    /* POSIX thread condvar (pthread_cond_*) implementation based on:
     * "Strategies for Implementing POSIX Condition Variables on Win32"
     * by Douglas C. Schmidt and Irfan Pyarali.
     */

Reply:
I'll go with `C11THREADS_NO_COND` for now and research the alternative implementation.